### PR TITLE
Fix Issue #1058

### PR DESCRIPTION
**Pull request by lupo1977**
Fix Issue #1058

### CODE REVIEW

File: ssh-keygen.c [hunk 1]
```diff
@@ -1541,12 +1541,9 @@ do_change_comment(struct passwd *pw)
...
 
 	strlcat(identity_file, ".pub", sizeof(identity_file));
-	fd = open(identity_file, O_WRONLY | O_CREAT | O_TRUNC, 0644);
-	if (fd == -1)
-		fatal("Could not save your public key in %s", identity_file);
-	f = fdopen(fd, "w");
+	f = fopen(identity_file, "w");
 	if (f == NULL)
-		fatal("fdopen %s failed: %s", identity_file, strerror(errno));
+		fatal("fopen %s failed: %s", identity_file, strerror(errno));
```
Comment (manojampalam) [inline ssh-keygen.c]: We cannot change core OpenSSH code. Add a #ifdef block.. Look elsewhere in this file that also had to deal with fdopen.

**Review by manojampalam — CHANGES_REQUESTED** [review]

**Comment by lupo1977** [comment]
Thanks for the hint. Hope it is ok now.

File: ssh-keygen.c [hunk 2]
```diff
@@ -1541,12 +1541,18 @@ do_change_comment(struct passwd *pw)
 	sshkey_free(private);
 
 	strlcat(identity_file, ".pub", sizeof(identity_file));
-	fd = open(identity_file, O_WRONLY | O_CREAT | O_TRUNC, 0644);
```
Comment (manojampalam) [inline ssh-keygen.c]: Understand this is a constructive change. But we prefer not to make any changes to base code to reduce merge conflicts. We are not anyway near to integrate this upstream.

**Review by manojampalam — CHANGES_REQUESTED** [review]

File: ssh-keygen.c (hunk 2 above)
Comment (lupo1977) [reply ssh-keygen.c]: Ok, got it.

**Comment by manojampalam** [comment]
Sorry, I did not realize this PR is in our shadow repo. Please create one on our main dev repo so that your contribution history wouldn't get lost - https://github.com/PowerShell/openssh-portable

**Comment by lupo1977** [comment]
Ok, got it. My mistake. Will raise another PR in the other branch.

**Pull request merged**